runners/google-cloud-dataflow-java/worker/src/main/java/org/apache/beam/runners/dataflow/worker/windmill/GrpcWindmillServer.java

```diff
@@ -987,6 +998,84 @@ public void append(StreamingGetWorkResponseChunk chunk) {
 
         this.data = data.concat(chunk.getSerializedWorkItem());
         this.bufferedSize += chunk.getSerializedWorkItem().size();
+        for (GetWorkStreamTimingInfo info : chunk.getPerWorkItemTimingInfosList()) {
```

Comment:
it might be nice to have a builder object that you just pass these timing infos to and then build/finalize to get the latencies. Then it would be easier to document and test independently too and we could experiment with keeping track differently.

Reply:
Done.